test/c3/appframework/appframework_test.go

```diff
@@ -211,4 +212,83 @@ var _ = Describe("c3appfw test", func() {
 		})
 	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("c3, appframework: can deploy a C3 SVA and have ES app installed on SHC", func() {
+
+			// ES is a huge file, we configure it here rather than in BeforeSuite/BeforeEach to save time for other tests
+			// Upload ES app to S3
+			appListV1 = testenv.ESApp
+			appFileList := testenv.GetAppFileList(appListV1, 1)
+
+			// Download V1 Apps from S3
+			err := testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileList)
+			Expect(err).To(Succeed(), "Unable to download V1 app files")
+
+			// Upload V1 apps to S3
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
+			Expect(err).To(Succeed(), "Unable to upload ES app to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Create App framework Spec
+			// volumeSpec: Volume name, Endpoint, Path and SecretRef
+			volumeName := "appframework-test-volume-" + testenv.RandomDNSName(3)
+			volumeSpec := []enterprisev1.VolumeSpec{testenv.GenerateIndexVolumeSpec(volumeName, testenv.GetS3Endpoint(), testenvInstance.GetIndexSecretName(), "aws", "s3")}
+
+			// AppSourceDefaultSpec: Remote Storage volume name and Scope of App deployment
+			appSourceDefaultSpec := enterprisev1.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   "cluster",
+			}
+
+			// appSourceSpec: App source name, location and volume name and scope from appSourceDefaultSpec
+			appSourceName := "appframework-" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterprisev1.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, s3TestDir, appSourceDefaultSpec)}
+
+			// appFrameworkSpec: AppSource settings, Poll Interval, volumes, appSources on volumes
+			appFrameworkSpec := enterprisev1.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			// Create Single site Cluster and SHC, with App Framework enabled on CM and SHC Deployer
+			shSpec := enterprisev1.SearchHeadClusterSpec{
+				CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+					Spec: splcommon.Spec{
+						ImagePullPolicy: "Always",
+					},
+					ExtraEnv: []corev1.EnvVar{
+						{
+							Name:  "SPLUNK_ES_SSL_ENABLEMENT",
+							Value: "ignore"},
+					},
+					Volumes:                      []corev1.Volume{},
+					LivenessInitialDelaySeconds:  1450,
+					ReadinessInitialDelaySeconds: 1450,
+				},
+				Replicas:           3,
+				AppFrameworkConfig: appFrameworkSpec,
+			}
+
+			_, err = deployment.DeploySearchHeadClusterWithGivenSpec(deployment.GetName()+"-shc", shSpec)
+			Expect(err).To(Succeed(), "Unable to deploy SHC with App framework")
+
+			// Ensure SHC go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify ES app is installed locally on SHC Deployer
```

Comment:
nit: I think its not installed locally on deployer, its just downloaded and is disabled, right?

Reply:
It has to be installed first to be disabled.
In ES doc, https://docs.splunk.com/Documentation/ES/6.4.1/Install/InstallEnterpriseSecuritySHC#Installing_Splunk_Enterprise_Security_from_the_command_line_in_a_search_head_cluster_environment , they mention "Start the installation process on the deployer".
If you log on deployer UI at some point you can see ES listed there.

Reply:
How do we manage to install it locally especially considering we have the scope set to "cluster"?